neovim/api/buffer.py

```diff
@@ -98,17 +100,34 @@ def range(self, start, end):
         return Range(self, start, end)
 
     def add_highlight(self, hl_group, line, col_start=0,
-                      col_end=-1, src_id=-1, async=None):
+                      col_end=-1, src_id=-1, async_=None,
+                      **kwargs):
         """Add a highlight to the buffer."""
-        if async is None:
-            async = (src_id != 0)
+        if async_ is None and 'async' in kwargs:
```

Comment:
This seems like it can be a helper function:
`async_ = check_async(async_, **kwargs, (src_id != 0))`

Reply:
Should be? It's just one line and used only here so I don't think it's a good idea to extract a simple code into a helper function.

Reply:
How so? it is 8-9 lines unrelated to the original meaning of the function in at least three places (here, `clear_highlight`, and `request`)

Reply:
I got. Sorry, I thought you are talking about exactly this line.

Reply:
Fixed 👍